tools/ctl/ctl.go

```diff
@@ -450,29 +450,22 @@ var runBenchmark = &cobra.Command{
 			// If we're talking to build-local directly, then we skip run-id generation.
 			run = time.Now().UTC().Format(time.RFC3339)
 		} else {
-			u := apiURL.JoinPath("runs")
-			values := url.Values{
-				"name": []string{filepath.Base(args[1])},
-				"hash": []string{hex.EncodeToString(set.Hash(sha256.New()))},
-				"type": []string{string(mode)},
-			}
-			u.RawQuery = values.Encode()
-			req, err := http.NewRequestWithContext(ctx, http.MethodPost, u.String(), nil)
-			if err != nil {
-				log.Fatal(errors.Wrap(err, "creating API version request"))
-			}
-			resp, err := client.Do(req)
+			resp, err := client.Do(makeHTTPRequest(ctx, apiURL.JoinPath("runs"), &schema.CreateRunRequest{
```

Comment:
Can you use a Stub here? I think that would get rid of the serialization boilerplate below as it could project all errors into the err return value on which you could unconditionally fail.

Reply:
Done.